### DISCUSSION TITLE
Migrate from java.util.logging (JUL) to SLF4J

### DISCUSSION THREAD

**Discussion by JP95Git** [bcgit/bc-java#1951] [Ideas]
How about migrating from java.util.logging (JUL) to SLF4J in Bouncy Castle Java (BC)?

I use BC in one of my projects and had to use the jul-to-slf4j bridge to hide certain logs of BC. Not a big deal, but all other dependencies either use no logging at all or SLF4J. So get rid of the jul-to-slf4j bridge in my project would be cool.

I also noticed some improvements when using SLF4J and Logback:
- My application runs much faster with the new logging.
- No more checking for log levels (if ...isLoggable...)
- Nice features like log rotation

**Comment by mwcw** [comment]
Hi,

Bouncy Castle is basically infrastructure, and it is important that we stick as much as possible to using what is in the standard libraries because we cannot make assumptions concerning how people may use BC.

With that in mind I do not support adding SLF4J to BC.

MW

**Comment by dghgit** [comment]
Yep, sorry but this is  not going to happen. We can't afford to introduce external dependencies.

**Comment by JP95Git** [comment]
Thanks for the very fast reply! This sounds like a legit reason for stick to JUL.

For the minority of projects who still do not use logging this is fine, because they just need BC not no other dependency.
But for the majority of projects who use logging, most of them use SLF4J, have to add another dependency (jul-to-slf4j bridge for example) if they want to use BC.

I have just noticed that you support JDK 1.1 (from 1997!). That's really impressive. SLF4J doesn't support such old JDKs at all, so that's probably another reason to stick with JUL.

Further readings:
https://stackoverflow.com/questions/11359187/why-not-use-java-util-logging
https://stackoverflow.com/questions/9740569/logger-for-java-library
https://stackoverflow.com/questions/7367103/best-way-to-set-up-logging-for-a-java-library-that-will-be-shared-among-many-pro
